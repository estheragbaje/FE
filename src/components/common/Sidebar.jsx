import React from 'react';
import styled from 'styled-components';

import ProfileImage from './ProfileImage';
import SideNavItem from './SideNavItem';
import {
  tabletMaxWidth, white, mediumSpace1, extraLargeSpace, smallSpace, largeSpace,
} from '../../styles/variables';
import { sidebarIcons, sidebarTexts } from '../../assets/icons/icons';


const SidebarContainer = () => (
  <StyledSidebar>
    <ProfileImage name="Bob Stone" image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ1m1DOmfCTW5yaJjjehDNcrTnJlFaph-ZADDYsCsyzwDJWS_6z" />
    <div className="nav-items">
      <SideNavItem text={sidebarTexts.surveys} icon={sidebarIcons.surveys} path="/surveys" />
      <SideNavItem
        text={sidebarTexts.settings}
        icon={sidebarIcons.settings}
        path="/profile"
      />
    </div>
    <div className="empty-div" />
  </StyledSidebar>
);

const StyledSidebar = styled.aside`
  min-height: 86vh;
  width: 15rem;
  padding: ${mediumSpace1} ${mediumSpace1};
  margin: ${largeSpace} 0 0 0;

  display: flex;
  flex-direction: column;
  background-color: ${white};

  .nav-items {
    padding: ${mediumSpace1} 0;
  }

  .empty-div {
    flex-grow: 1;
  }

  .grey-logo {
    margin: 0 auto;
    width: 70%;

    img {
      width: 100%;
    }
  }

  @media only screen and (max-width: ${tabletMaxWidth}) {
    display: none;
  }
`;

export default SidebarContainer;
