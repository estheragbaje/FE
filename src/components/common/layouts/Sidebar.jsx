import React from 'react';
import styled from 'styled-components';

import ProfileImage from '../ProfileImage';
import SideNavItem from './SideNavItem';
import {
  tabletMaxWidth,
  white,
  mediumSpace1,
} from '../../../styles/variables';
import { sidebarIcons, sidebarTexts } from '../../../assets/icons/icons';

const SidebarContainer = () => (
  <StyledSidebar>
    <ProfileImage
      name="Bob Stone"
      image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ1m1DOmfCTW5yaJjjehDNcrTnJlFaph-ZADDYsCsyzwDJWS_6z"
    />
    <div className="nav-items">
      <SideNavItem
        text={sidebarTexts.surveys}
        icon={sidebarIcons.surveys}
        path="/dashboard"
      />
      <SideNavItem
        text={sidebarTexts.settings}
        icon={sidebarIcons.settings}
        path="/profile"
      />
    </div>
    <div className="empty-div" />
  </StyledSidebar>
);

const StyledSidebar = styled.aside`
	min-height: calc(100vh - 100px);
	width: 15rem;

	display: flex;
	flex-direction: column;
	background-color: ${white};

	.nav-items {
		padding: ${mediumSpace1} 0;
	}

	.empty-div {
		flex-grow: 1;
	}

	.grey-logo {
		margin: 0 auto;
		width: 70%;

		img {
			width: 100%;
		}
	}

	@media only screen and (max-width: ${tabletMaxWidth}) {
		display: none;
	}
`;

export default SidebarContainer;
