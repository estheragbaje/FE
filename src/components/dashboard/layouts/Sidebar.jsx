import React from 'react';
import jwtDecode from 'jwt-decode';
import ProfileImage from '../../common/ProfileImage';
import SideNavItem from './SideNavItem';
import { sidebarIcons, sidebarTexts } from '../../../assets/icons/icons';
import { StyledSidebar } from './DashboardLayoutStyles';
import { MyContext } from '../../../Provider';
import avatar from '../../../assets/images/avatar-default.png';

class SidebarContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: null,
    };
  }

  componentDidMount() {
    const token = localStorage.getItem('token');
    if (token) {
      const user = jwtDecode(token);
      this.setState({ user });
    } else {
      this.setState({ user: null });
    }
  }

  render() {
    const { user } = this.state;
    return (
      <MyContext.Consumer>
        {(context) => (
          <StyledSidebar className={context.sidebarOpen ? 'active' : ''}>
            <ProfileImage
              name={user && user.username}
              image={avatar}
            />
            <div
              className="nav-items"
              role="presentation"
              onClick={context.toggleSidebar}
            >
              <SideNavItem
                text={sidebarTexts.create_survey}
                icon={sidebarIcons.create_survey}
                path="/create_survey"
              />

              <div id="logout" role="presentation" onClick={context.logout}>
                <SideNavItem
                  text="Logout"
                  icon={sidebarIcons.logout}
                  path="/"
                />
              </div>
            </div>
            <div className="empty-div" />
          </StyledSidebar>
        )}
      </MyContext.Consumer>
    );
  }
}

export default SidebarContainer;
