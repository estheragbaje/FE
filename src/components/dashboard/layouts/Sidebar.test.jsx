import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from './Sidebar';

describe('<Sidebar />', () => {
  it('renders the Sidebar component correctly', () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper).toMatchSnapshot();
  });
});
