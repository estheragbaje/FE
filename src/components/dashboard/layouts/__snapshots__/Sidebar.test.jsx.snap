// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders the Sidebar component correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
